pkg/storage/store.go

```diff
@@ -183,7 +186,8 @@ func (s *store) init() error {
 			return err
 		}
 
-		s.composite.AddStore(p.From.Time, f, idx, w, stop)
+		// s.Store is always assigned the CompositeStore implementation of the Store interface
+		s.Store.(*stores.CompositeStore).AddStore(p.From.Time, f, idx, w, stop)
```

Comment:
alternate approach: add `CompositeStore` interface in store.go that defines `AddStore` contract. replace `composite *stores.CompositeStore` with the new interface `composite CompositeStore`. wdyt?

Reply:
if we could remove all usages of `storage.Store` interface eventually, `CompositeStore` can be embedded within the `Store` interface itself without the need for an extra field

Reply:
`s.Store` may be replaced by the AsyncStore https://github.com/grafana/loki/blob/main/pkg/storage/store.go#L189-L191